include/aspect/assembly.h

```diff
@@ -59,6 +60,8 @@ namespace aspect
 
           std::vector<SymmetricTensor<2,dim> > grads_phi_u;
           std::vector<double>                  phi_p;
+          std::vector<double>                  phi_p_c;
+          std::vector<Tensor<1,dim> >          grad_phi_p;
```

Comment:
I'm really not excited about adding the gradient of the pressure. The pressure is in L2, and so we cannot expect the gradient of the pressure to be well approximated at all. 

In particular, the pressure shape functions may be discontinuous (if the "conservative" formulation is chosen in the input file), and in that case the gradient of the pressure really makes no sense. Can you, at least, add an assertion somewhere that checks that this is not the case?